webapp/src/components/modals/attach_comment_to_issue/attach_comment_to_issue.jsx

```diff
@@ -0,0 +1,241 @@
+import {Modal} from 'react-bootstrap';
+
+import JiraFields from 'components/jira_fields';
+import FormButton from 'components/form_button';
+import Loading from 'components/loading';
+import ReactSelectSetting from 'components/react_select_setting';
+import Input from 'components/input';
+
+import {getProjectValues, getFields} from 'jira_issue_metadata';
+
+const initialState = {
+    submitting: false,
+    projectKey: null,
+    issueKey: null,
+
+    fields: {
+        ID: '',
+        description: '',
+        project: {
+            key: '',
+        },
+    },
+    error: null,
+};
+
+export default class AttachIssueModal extends PureComponent {
+    static propTypes = {
+        close: PropTypes.func.isRequired,
+        create: PropTypes.func.isRequired,
+        post: PropTypes.object,
+        theme: PropTypes.object.isRequired,
+        visible: PropTypes.bool.isRequired,
+        jiraIssueMetadata: PropTypes.object,
+        fetchJiraIssueMetadata: PropTypes.func.isRequired,
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = initialState;
+    }
+
+    componentDidUpdate(prevProps) {
+        if (this.props.post && (!prevProps.post || this.props.post.id !== prevProps.post.id)) {
+            this.props.fetchJiraIssueMetadata();
+            const fields = {...this.state.fields};
+            fields.description = this.props.post.message;
+            this.setState({fields}); //eslint-disable-line react/no-did-update-set-state
+        }
+    }
+
+    handleCreate = (e) => {
+        if (e && e.preventDefault) {
+            e.preventDefault();
+        }
+
+        const issue = {
+            post_id: this.props.post.id,
+            fields: this.state.fields,
+        };
+
+        this.setState({submitting: true});
+
+        this.props.create(issue).then((created) => {
+            if (created.error) {
+                this.setState({error: created.error.message, submitting: false});
+                return;
+            }
+            this.handleClose(e);
+        });
+    };
+
+    handleClose = (e) => {
+        if (e && e.preventDefault) {
+            e.preventDefault();
+        }
+        const {close} = this.props;
+        this.setState(initialState, close);
+    };
+
+    handleDescriptionChange = (e) => {
+        const description = e.target.value;
+        const {fields} = this.state;
+        const nFields = {
+            ...fields,
+            description,
+        };
+
+        this.setState({fields: nFields});
+    };
+
+    handleProjectChange = (id, value) => {
+        const fields = {...this.state.fields};
+        const projectKey = value;
+        fields.project = {
+            key: value,
+        };
+
+        this.setState({
+            projectKey,
+            fields,
+        });
+    }
+
+    handleIssueKeyChange = (id, value) => {
+        const fields = {...this.state.fields};
+        const issueKey = value;
+
+        fields.ID = value;
```

Comment:
Why set this on fields.id and not just issueKey?

Reply:
Agree and clearing this up in the next release.  Thanks @crspeller